RevenueCatUI/CustomerCenter/Data/PurchaseInformation.swift

```diff
@@ -123,6 +136,115 @@ struct PurchaseInformation {
 }
 // swiftlint:enable nesting
 
+extension PurchaseInformation {
+
+    /// Provides detailed information about a user's purchase, including entitlement and renewal price.
+    ///
+    /// This function fetches the renewal price details for the given product asynchronously from
+    /// StoreKit 2 and constructs a `PurchaseInformation` object with the provided
+    /// transaction, entitlement, and subscribed product details.
+    ///
+    /// - Parameters:
+    ///   - entitlement: Optional entitlement information associated with the purchase.
+    ///   - subscribedProduct: The product the user has subscribed to, represented as a `StoreProduct`.
+    ///   - transaction: The transaction information for the purchase.
+    /// - Returns: A `PurchaseInformation` object containing the purchase details, including the renewal price.
+    ///
+    /// - Availability: iOS 15.0+
+    @available(iOS 15.0, *)
+    @available(macOS, unavailable)
+    @available(tvOS, unavailable)
+    @available(watchOS, unavailable)
+    static func purchaseInformationUsingRenewalInfo(
+        entitlement: EntitlementInfo? = nil,
+        subscribedProduct: StoreProduct,
+        transaction: Transaction
+    ) async -> PurchaseInformation {
+        let renewalPriceDetails = await Self.extractPriceDetailsFromRenwalInfo(forProduct: subscribedProduct)
+        return PurchaseInformation(
+            entitlement: entitlement,
+            subscribedProduct: subscribedProduct,
+            transaction: transaction,
+            renewalPrice: renewalPriceDetails
+        )
+    }
+
+    @available(iOS 15.0, *)
+    @available(macOS, unavailable)
+    @available(tvOS, unavailable)
+    @available(watchOS, unavailable)
+    private static func extractPriceDetailsFromRenwalInfo(
+        forProduct product: StoreProduct
+    ) async -> PriceDetails? {
+        if #available(macOS 12.0, tvOS 15.0, *) {
```

Comment:
Despite having these functions marked as unavailable on these platforms, the compiler still requires me to have availability checks to avoid compilation errors 🙈